libraries/Ethernet/src/ETH.cpp

```diff
@@ -19,6 +19,10 @@
  */
 
 // Disable the automatic pin remapping of the API calls in this file
+
+#include "sdkconfig.h"
+#if CONFIG_ETH_ENABLED
```

Comment:
this check can be after `#include "ETH.h" and then you do not need to include sdkconfig again (already included by ETH.h.